functions.go

```diff
@@ -8,6 +8,16 @@ type Function struct {
 	GoName     string // Go name of the function with the API prefix stripped
 	Parameters []Parameter
 	Return     Type
+	Overloads  []Overload
+}
+
+// An Overload describes an alternative signature for the same function.
+type Overload struct {
+	Name         string // C name of the original function
+	GoName       string // Go name of the original function
```

Comment:
could this value be pulled from the parent struct? similarly for Name?

Reply:
`Name` turned out not to be necessary. As for `GoName`, I was not able to figure out an easy way to access `GoName` from the parent from the sub-template `overloadCall`. Hence I kept this field